src/main/java/ru/vidtu/ias/IAS.java

```diff
@@ -0,0 +1,307 @@
+     * @param version       Mod version
+     * @param loader        Mod loader
+     * @param loaderVersion Mod loader version
+     * @param gameVersion   Game version
+     */
+    public static void init(Path gamePath, Path configPath, String version, String loader, String loaderVersion, String gameVersion) {
+        // Initialize the dirs.
+        gameDirectory = gamePath;
+        configDirectory = configPath;
+
+        // Set up IAS.
+        userAgent = USER_AGENT_TEMPLATE.formatted(version, SESSION, loader, loaderVersion, gameVersion, Runtime.version().toString());
+        LOGGER.info("IAS user agent: {}", userAgent);
+
+        // Write the disclaimers.
+        try {
+            disclaimersStorage();
+        } catch (Throwable t) {
+            LOGGER.error("IAS: Unable to write disclaimers.", t);
+        }
+
+        // Read the config.
+        try {
+            loadConfig();
+        } catch (Throwable t) {
+            LOGGER.error("IAS: Unable to load IAS config.", t);
+        }
+
+        // Read the storage.
+        try {
+            loadStorage();
+        } catch (Throwable t) {
+            LOGGER.error("IAS: Unable to load IAS storage.", t);
+        }
+
+        // Create the executor.
+        executor = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "IAS Executor Thread"));
+
+        // Perform initial loading.
+        if (Boolean.getBoolean("ias.skipDisableScanning")) return;
+        executor.scheduleWithFixedDelay(() -> {
+            // Perform scanning, if allowed.
+            try {
+                // Skip if not allowed or already disabled.
+                if (disabled || Boolean.getBoolean("ias.skipDisableScanning")) return;
+
+                // Create the client.
+                HttpClient client = HttpClient.newBuilder()
+                        .connectTimeout(TIMEOUT)
+                        .version(HttpClient.Version.HTTP_2)
+                        .followRedirects(HttpClient.Redirect.NORMAL)
+                        .executor(Runnable::run)
+                        .priority(1)
+                        .build();
+
+                // Send the request.
+                HttpResponse<Stream<String>> response = client.send(HttpRequest.newBuilder()
+                        .uri(new URI("https://raw.githubusercontent.com/The-Fireplace-Minecraft-Mods/In-Game-Account-Switcher/main/.ias/disabled_v1"))
+                        .header("User-Agent", userAgent)
+                        .timeout(TIMEOUT)
+                        .GET()
+                        .build(), HttpResponse.BodyHandlers.ofLines());
+
+                // Validate the code.
+                int code = response.statusCode();
+                if (code < 200 || code > 299) return;
+
+                // Check the lines.
+                disabled = response.body().anyMatch(line -> {
+                    line = line.strip();
+                    return "ALL".equalsIgnoreCase(line) || version.equalsIgnoreCase(line);
```

Comment:
In what situation do you foresee the need to disable ALL versions?

Reply:
in a situation, where, for some bizarre reason, versioning or publishing will go nuts